src/ui/Search.js

```diff
@@ -0,0 +1,148 @@
+import React, {Component} from 'react';
+import Modal from 'react-modal';
+import {UP, DOWN, ESC, ENTER, PGUP, PGDN, F3} from '../keycode';
+
+// TODO: Make sure to bind modal to your appElement (http://reactcommunity.org/react-modal/accessibility/)
+// Modal.setAppElement('#yourAppElement');
+
+export default class extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {showSearchModal: false, searchString: '', searchMode: 0};
+    this.searchBoxNode = null;
+  }
+
+  handleKeyModal = e => {
+    if (e.keyCode === ENTER || e.keyCode === ESC) { // enter or escape
+      this.setState({showSearchModal: false});
```

Comment:
We should call into `this.blocks.sayAndForget` to notify the user that the modal has been closed.